### PR TITLE
Cyberphys/hacker/fixes

### PR DESCRIPTION
**Pull request by lolsborn**

### CODE REVIEW

File: besspin/cyberPhys/ui/hacker-kiosk/zmq_python_test.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import time
+import json
+import random
+
+context = zmq.Context()
+socket = context.socket(zmq.PAIR)
+socket.bind("tcp://*:3333")
+
+while True:
+    message = socket.recv()
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/zmq_python_test.py:13]: you can ditch `import json` completely and use `socket.recv_json()`

File: besspin/cyberPhys/ui/hacker-kiosk/src/background.js [hunk 2]
```diff
@@ -25,6 +27,31 @@ ipcMain.on('can-poll', (event) => {
   }
 })
 
+// ZMQ Network
+let zmq_sock = zmq.socket("pair");
+let zmq_address = "tcp://127.0.0.1:3333";
+zmq_sock.connect(zmq_address);
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/background.js:33]: Do you want to make this compatible with the Python interface in this PR or in a subsequent PR?

File: besspin/cyberPhys/ui/hacker-kiosk/src/components/Hack01_Start.vue [hunk 3]
```diff
@@ -54,18 +55,32 @@
...
     },
     mounted() {
     },
     methods: {
+      disableTimer() {
+        clearInterval(globalTimer);
+        this.next();
+      },
+      resetTimer() {
+        globalTimer = setInterval(() => { console.log('reset'); window.location = '/#/hack02_intro';}, 5 * 60 * 1000);
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/components/Hack01_Start.vue:75]: Perhaps better to use a global variable/config file?

File: besspin/cyberPhys/ui/hacker-kiosk/src/background.js [hunk 4]
```diff
@@ -25,6 +27,31 @@ ipcMain.on('can-poll', (event) => {
   }
 })
 
+// ZMQ Network
+let zmq_sock = zmq.socket("pair");
+let zmq_address = "tcp://127.0.0.1:3333";
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/background.js:32]: The connection port could be added into a global config file.

File: besspin/cyberPhys/ui/hacker-kiosk/src/components/Hack05_InfoAttempt.vue [hunk 5]
```diff
@@ -28,18 +27,44 @@
...
       }
     },
     mounted() {
+      let vm = this;
+      ipc.on('zmq-results',(event, q) => {
+        q.forEach(item => {
+          console.log("item", item);
+          //TODO: Handle Failure?
+          if(item.function == "hack-info" && item.status == 200) {
+            vm.$router.push({ name: 'hack06_info_exploit' });
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/components/Hack05_InfoAttempt.vue:51]: The syntax is:

* Request:  `{func: "function to be called" args: {arguments here, can be an empty dictionary}}`
* Response: `{func: "function that was called" args: {arguments used can be an empty dictionary}, retval: "some-return-string", status:  response-status-code}`
```
Exchange format:
{
    func: "name-of-the-function-to-be-called",
    args: {"dictionary of arguments, can be an empty dictionary}
    resp: {"dictionary of return vaues, can be an empty dictionary}
    retval: "return value, assumed to be a string"
    status: response status (integer)
        200 -OK
        500 - unexpected failure
        501 - func not implemented
}
```

The function calls are not important right now, I can easily add them at the end.

File: besspin/cyberPhys/ui/hacker-kiosk/zmq_python_test.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+
+import zmq
+import time
+import json
+import random
+
+context = zmq.Context()
+socket = context.socket(zmq.PAIR)
+socket.bind("tcp://*:3333")
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/zmq_python_test.py:10]: I think we can remove this file and use https://github.com/GaloisInc/BESSPIN-Tool-Suite/blob/develop/besspin/cyberPhys/hackerKiosk/hackerKiosk.py as your backend.

File: besspin/cyberPhys/ui/hacker-kiosk/src/background.js [hunk 7]
```diff
@@ -5,10 +5,12 @@ import { createProtocol } from 'vue-cli-plugin-electron-builder/lib'
 import installExtension, { VUEJS_DEVTOOLS } from 'electron-devtools-installer'
 const isDevelopment = process.env.NODE_ENV !== 'production'
 import {CanListener, CanNetwork} from '../../common/can';
+import zmq, { socket } from 'zeromq';
 
 
 let canQueue= [];
 
+// UDP Network
 let can_net = new CanNetwork(5002);
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/background.js:14]: Why does the kiosk need to listen to the CAN network?

File: besspin/cyberPhys/ui/hacker-kiosk/src/background.js [hunk 8]
```diff
@@ -25,6 +27,31 @@ ipcMain.on('can-poll', (event) => {
...
+  console.log("zmq message recieved: ", decoded);
+  zmqQueue.push(decoded);
+});
+
+ipcMain.on('zmq-poll', (event) => {
+  event.reply('zmq-results',  JSON.parse(JSON.stringify(zmqQueue)));
+  zmqQueue = [];
+});
+
+ipcMain.on('cmd-msg', (event, cmd_id) => {
```
Comment (podhrmic) [inline besspin/cyberPhys/ui/hacker-kiosk/src/background.js:48]: I think this needs to be updated to reflect the JSON exchange format (see my comment below).

**Review by podhrmic — CHANGES_REQUESTED** [review]

File: besspin/cyberPhys/ui/hacker-kiosk/src/background.js (hunk 7 above)
Comment (dmzimmerman) [reply besspin/cyberPhys/ui/hacker-kiosk/src/background.js:14]: I don't think it should; it needs to listen to the special "hacked" CAN network that the GPS packets come in on, but other than that I don't think it uses CAN packets for anything.

**Review by podhrmic — APPROVED** [review]

**Comment by podhrmic** [comment]
Merging as is, will clean it up in a new PR

**Pull request merged**